### PR TITLE
Feat/lw 7585 implement serialization of witness set

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

In order to use the SDK in the mobile environment is now a priority to replace the CML library with our custom serialization classes.

# Proposed Solution

Implement the serialization natively in Typescript.

# Important Changes Introduced

- Added transaction witness set serialization classes.

### CODE REVIEW

File: packages/core/src/Serialization/TransactionWitnessSet/TransactionWitnessSet.ts [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+  native: Array<NativeScript> | undefined;
+  plutusV1: Array<PlutusV1Script> | undefined;
+  plutusV2: Array<PlutusV2Script> | undefined;
+};
+
+/**
+ * A witness is a piece of information that allows you to efficiently verify the
+ * authenticity of the transaction (also known as proof).
+ *
+ * In Cardano, transactions need multiple types of authentication proofs, these can range
```
Comment (rhyslbw) [inline packages/core/src/Serialization/TransactionWitnessSet/TransactionWitnessSet.ts:27]:
```suggestion
 * In Cardano, transactions have multiple types of authentication proofs, these can range
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Awesome work @AngelCastilloB 🔥 

It looks like 0f7298aef7cea11973f973398979bff61e9e2940 should be squashed with the first commit though, if the tests were failing without it. It's great to see this implementation has a positive effect on transaction cost 😎

File: packages/tx-construction/test/fees/fees.test.ts [hunk 2]
```diff
@@ -120,7 +120,7 @@ describe('fees', () => {
     it('calculate the correct min fee for transaction with scripts', () => {
       const prices = { memory: 0.0577, steps: 0.000_007_21 };
       const fee = minFee(babbageTx, prices, MinFeeConstant(155_381), MinFeeCoefficient(44));
-      expect(fee).toBe(218_807n);
+      expect(fee).toBe(218_763n);
```
Comment (mkazlauskas) [inline packages/tx-construction/test/fees/fees.test.ts:123]: Witness set is now smaller? :mage:

**Review by mkazlauskas — APPROVED** [review]
Great work!

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**